gbpservice/neutron/nfp/configurator/api/v1/controllers/controller.py

```diff
@@ -0,0 +1,130 @@
+import json
+import subprocess
+
+from neutron.common import rpc as n_rpc
+from neutron.agent.common import config
+from oslo_config import cfg
+from oslo_log import log as logging
+import oslo_messaging
+from pecan import expose, request, response
+from pecan import rest
+
+
+import constants
+
+LOG = logging.getLogger(__name__)
+
+"""Controller class for handling all the HTTP requests.
+This Controller class serves all the HTTP requests sent by
+config-agent to configurator and make RPC call accordingly."""
+
+
+class Controller(rest.RestController):
+
+    def __init__(self, method_name):
+        try:
+            self.host = subprocess.check_output(
+                            'hostname', shell=True).rstrip()
+        except Exception as err:
+            msg = ("Failed to get hostname  %s." % str(err).capitalize())
+            LOG.error(msg)
+        self.rpcclient = RPCClient(topic=constants.TOPIC, host=self.host)
+        self.method_name = method_name
+        super(Controller, self).__init__()
+
+    @expose(method='GET', content_type='application/json')
+    def get(self):
+        """get method of REST server.
+        This method returns Notification data to config-agent"""
+        try:
+            notification_data = json.dumps(self.rpcclient.call())
+            msg = ("NOTIFICATION_DATA sent to config_agent %s"
+                   % notification_data)
+            LOG.info(msg)
+            return notification_data
+        except Exception as err:
+            pecan.response.status=400
+            msg = ("Failed to get notification_data  %s."
+                   % str(err).capitalize())
+            LOG.error(msg)
+            return json.dumps({'ERROR': msg})
+
+    @expose(method='POST', content_type='application/json')
+    def post(self, **body):
+        """post method REST server.
+        cast RPC according to the post request by using
+        object of RPCClient"""
+        try:
+            body = None
+            if request.is_body_readable:
+                body = request.json_body
+            
+            self.rpcclient.cast(self.method_name, body)
+            msg = ("Successfully served HTTP request %s" % self.method_name)
+            LOG.info(msg)
+            return json.dumps({'SUCCESS': self.method_name})
+        except Exception as err:
+            pecan.response.status=400
+            msg = ("Failed to serve HTTP post request %s %s."
+                   % (self.method_name, str(err).capitalize()))
+            LOG.error(msg)
+            return json.dumps({'ERROR': msg})
```

Comment:
Change the return data in case of error as following.
"failure_desc": {
    "msg": "Detailed Error Message"
}